src/metabase/lib/filter/operator.cljc

```diff
@@ -0,0 +1,201 @@
+   (operator-def :is-null :is-empty)
+   (operator-def :not-null :not-empty)])
+
+(def ^:private location-operators
+  [(operator-def :=)
+   (operator-def :!=)
+   (operator-def :is-empty)
+   (operator-def :not-empty)
+   (operator-def :contains)
+   (operator-def :does-not-contain)
+   (operator-def :starts-with)
+   (operator-def :ends-with)])
+
+(def ^:private temporal-operators
+  [(operator-def :!= :excludes)
+   (operator-def :=)
+   (operator-def :< :before)
+   (operator-def :> :after)
+   (operator-def :between)
+   (operator-def :is-null :is-empty)
+   (operator-def :not-null :not-empty)])
+
+(def ^:private coordinate-operators
+  [(operator-def :=)
+   (operator-def :!=)
+   (operator-def :inside)
+   (operator-def :>)
+   (operator-def :<)
+   (operator-def :between)
+   (operator-def :>=)
+   (operator-def :<=)])
+
+(def ^:private number-operators
+  [(operator-def := :equal-to)
+   (operator-def :!= :not-equal-to)
+   (operator-def :>)
+   (operator-def :<)
+   (operator-def :between)
+   (operator-def :>=)
+   (operator-def :<=)
+   (operator-def :is-null :is-empty)
+   (operator-def :not-null :not-empty)])
+
+(def ^:private text-operators
+  [(operator-def :=)
+   (operator-def :!=)
+   (operator-def :contains)
+   (operator-def :does-not-contain)
+   (operator-def :is-null)
+   (operator-def :not-null)
+   (operator-def :is-empty)
+   (operator-def :not-empty)
+   (operator-def :starts-with)
+   (operator-def :ends-with)])
+
+(def ^:private text-like-operators
+  [(operator-def :=)
+   (operator-def :!=)
+   (operator-def :is-null)
+   (operator-def :not-null)
+   (operator-def :is-empty)
+   (operator-def :not-empty)])
+
+(def ^:private boolean-operators
+  [(operator-def :=)
+   (operator-def :is-null :is-empty)
+   (operator-def :not-null :not-empty)])
+
+(def ^:private default-operators
+  [(operator-def :=)
+   (operator-def :!=)
+   (operator-def :is-null)
+   (operator-def :not-null)])
+
+(def join-operators
+  "Operators that should be listed as options in join conditions."
+  [(assoc (operator-def :=) :default true)
+   (operator-def :>)
+   (operator-def :<)
+   (operator-def :>=)
+   (operator-def :<=)
+   (operator-def :!=)])
+
+(mu/defn filter-operators :- [:sequential ::lib.schema.filter/operator]
+  "The list of available filter operators.
+   The order of operators is relevant for the front end.
+   There are slight differences between names and ordering for the different base types."
+  [column :- lib.metadata/ColumnMetadata]
+  ;; The order of these clauses is important since we want to match the most relevant type
+  (condp #(lib.types.isa/isa? %2 %1) column
+    :type/PK         key-operators
+    :type/FK         key-operators
+    :type/Location   location-operators
+    :type/Temporal   temporal-operators
+    :type/Coordinate coordinate-operators
+    :type/Number     number-operators
+    :type/Text       text-operators
+    :type/TextLike   text-like-operators
+    :type/Boolean    boolean-operators
```

Comment:
I've changed `lib.types.isa/isa?` to `lib.types.isa/field-type?` in order to fix a bug (see #32067).